src/uu/install/src/install.rs

```diff
@@ -408,98 +488,85 @@ fn standard(mut paths: Vec<String>, b: Behavior) -> i32 {
-                "invalid target {}: No such file or directory",
-                target.display()
-            );
-            1
+            Err(InstallError::InvalidTarget(target).into())
         }
     }
 }
 
 /// Copy some files into a directory.
 ///
 /// Prints verbose information and error messages.
-/// Returns an integer intended as a program return code.
+/// Returns a Result type with the Err variant containing the error message.
 ///
 /// # Parameters
 ///
 /// _files_ must all exist as non-directories.
 /// _target_dir_ must be a directory.
 ///
-fn copy_files_into_dir(files: &[PathBuf], target_dir: &Path, b: &Behavior) -> i32 {
+fn copy_files_into_dir(files: &[PathBuf], target_dir: &Path, b: &Behavior) -> UResult<()> {
     if !target_dir.is_dir() {
-        show_error!("target '{}' is not a directory", target_dir.display());
-        return 1;
+        return Err(InstallError::TargetDirIsntDir(target_dir.to_path_buf()).into());
     }
-
-    let mut all_successful = true;
     for sourcepath in files.iter() {
         if !sourcepath.exists() {
-            show_error!(
-                "cannot stat '{}': No such file or directory",
-                sourcepath.display()
+            let err = UIoError::new(
+                std::io::ErrorKind::NotFound,
+                format!("cannot stat '{}'", sourcepath.display()),
             );
-
-            all_successful = false;
+            show!(err);
             continue;
         }
 
         if sourcepath.is_dir() {
-            show_error!("omitting directory '{}'", sourcepath.display());
-            all_successful = false;
+            let err = InstallError::OmittingDirectory(sourcepath.to_path_buf());
+            show!(err);
             continue;
         }
 
         let mut targetpath = target_dir.to_path_buf();
         let filename = sourcepath.components().last().unwrap();
         targetpath.push(filename);
 
-        if copy(sourcepath, &targetpath, b).is_err() {
-            all_successful = false;
+        if let Err(e) = copy(sourcepath, &targetpath, b) {
+            show!(e);
         }
     }
-    if all_successful {
-        0
-    } else {
-        1
-    }
+    // If the exit code was set, or show! has been called at least once
+    // (which sets the exit code as well), function execution will end after
+    // this return.
+    Ok(())
 }
 
 /// Copy a file to another file.
 ///
 /// Prints verbose information and error messages.
-/// Returns an integer intended as a program return code.
+/// Returns a Result type with the Err variant containing the error message.
 ///
 /// # Parameters
 ///
 /// _file_ must exist as a non-directory.
 /// _target_ must be a non-directory
 ///
-fn copy_file_to_file(file: &Path, target: &Path, b: &Behavior) -> i32 {
-    if copy(file, target, b).is_err() {
-        1
-    } else {
-        0
-    }
+fn copy_file_to_file(file: &Path, target: &Path, b: &Behavior) -> UResult<()> {
+    copy(file, target, b)
```

Comment:
I think this function is now useless :)

Reply:
Well spotted, I completely missed that one. Thanks!